pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader_test.go

```diff
@@ -287,6 +287,18 @@ func (it *fakeChainReaderInterfaceTester) GetChainReader(_ *testing.T) types.Con
 	return it.impl
 }
 
+func (it *fakeChainReaderInterfaceTester) StartChainReader(t *testing.T) {
```

Comment:
Highlights of the file:

- Breaking change within `ChainReaderInterfaceTester` to add `StartChainReader` and `CloseChainReader`. 
- Add `isStarted` flag to check if the service `isStarted`.
- Add `fakeChainReader` implementations of `Start` and `Close`. Add `IsStarted` checks within `GetLatestValue`, `BatchGetLatestValues`, and `QueryKey`.